### DISCUSSION TITLE
REGRESSION in renovate v36 for config.js when referencing environment variables via process.env.*

### DISCUSSION THREAD

**Discussion by abjoseph** [renovatebot/renovate#23257] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

v36.3.0

### If you're self-hosting Renovate, select which platform you are using.

Gitea or Forgejo

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

The below config worked perfectly fine with v35, however upon updating to v36 renovate validation for my config.js file no longer passes and constantly fails with the error below.

Also, please ***note***, there are no issues with how I'm passing the env variables as if I simply revert back to v35 with no other changes, it works as expected.

Please see the contents of my `config.js` file below: (obviously with domain names obfuscated)

```js
module.exports = {
  platform: 'gitea',
  endpoint: 'https://git.example.org/api/v1/',
  gitAuthor: 'Renovate Bot <svc_renovate@example.org>',
  username: 'svc_renovate',
  autodiscover: true,
  onboardingConfig: {
    $schema: 'https://docs.renovatebot.com/renovate-schema.json',
    extends: ['config:recommended']
  },
  optimizeForDisabled: true,
  persistRepoData: true,
  repositories: [
    "owner/example-repository"
  ],
  exposeAllEnv: true,
  hostRules: [
    {
      hostType: 'docker',
      matchHost: 'docker.io',
      username: 'dockerhub_user',
      password: process.env.DOCKER_HUB_PASSWORD,
    },
    {
      hostType: 'docker',
      matchHost: 'registry.example.org',
      username: 'robot$serviceaccount',
      password: process.env.HARBOR_REGISTRY_PASSWORD,
    },
  ]
}
```

### Relevant debug logs

<details><summary>Logs</summary>

```bash
+ unset GIT_COMMITTER_NAME GIT_COMMITTER_EMAIL GIT_AUTHOR_NAME GIT_AUTHOR_EMAIL
+ renovate-config-validator
DEBUG: Using RE2 as regex engine
 INFO: Validating renovate.json
DEBUG: Checking for config file in /drone/src/config.js
 INFO: Validating config.js
ERROR: Found errors in configuration
       "file": "config.js",
       "errors": [
         {
           "topic": "Configuration Error",
           "message": "Configuration option `hostRules[0].password` should be a string"
         },
         {
           "topic": "Configuration Error",
           "message": "Configuration option `hostRules[1].password` should be a string"
         }
       ]
```

</details>

### Have you created a minimal reproduction repository?

Placeholder value, please select the correct response from the dropdown

**Comment by rarkins** [comment]
I cannot reproduce this locally

**Reply by abjoseph** [reply]
Can you share what CI you used for your repro and whether or not you used the renovate container or the cli binary directly?

**Comment by viceice** [comment]
you probably forgot to set the ENV values when running the validator, so that strings are empty?

**Reply by abjoseph** [reply]
> Also, please note, there are no issues with how I'm passing the env variables as if I simply revert back to v35 with no other changes, it works as expected.

**Comment by rarkins** [comment]
Describe exactly how you run renovate (environment, commands, image, etc)
